### PR TITLE
Fix MON_IP in the Ceph pod for multinode OCP CI

### PR DESCRIPTION
**Pull request by fmount**
The `Ceph` pod is supposed to be single node, single pod, single container. If `MON_IP` is not passed and computed by the `get_pod_ip` function, in a `multinode` context it might return a list of `InternalIP` addresses. This patch represents an attempt to only support a single `IP` address from the list and assign it to the `mon` spawned as first process during the cluster bootstrap.
By doing this we pin the ceph pod on a specific (worker) node using `nodeSelector`.

### CODE REVIEW

**Review by fultonj — COMMENTED** [review]
+2

**Comment by Sandeepyadav93** [comment]
recheck

**Comment by Sandeepyadav93** [comment]
/lgtm

thank you, Tested locally on multinode OCP environment, it works.

`make ceph` target is only for dev/ci usecase, so I think its okay to pin ceph pod to a particular worker.

File: scripts/gen-ceph-kustomize.sh [hunk 1]
```diff
@@ -119,7 +122,12 @@ function get_pod_ip {
         echo "HOSTNETWORK must be set to true!"
         exit 1
     fi
-    MON_IP=$(oc get nodes -o 'jsonpath={.items[*].status.addresses[?(.type=="InternalIP")].address}')
+    NODES=$(oc get nodes -o 'jsonpath={.items[*].status.addresses[*].address}')
```
Comment (Sandeepyadav93) [inline scripts/gen-ceph-kustomize.sh:125]: @fmount I think we should only filter worker nodes

oc get nodes --selector='node-role.kubernetes.io/worker='

Comment (fmount) [reply scripts/gen-ceph-kustomize.sh:125]: +1, doing that in my next update.

Comment (fmount) [reply scripts/gen-ceph-kustomize.sh]: Done

**Comment by Sandeepyadav93** [comment]
/lgtm

thanks for the update

**Pull request merged**